File: apps/server/aetherblog-service/blog-service/src/main/java/com/aetherblog/blog/entity/MediaVersion.java
package com.aetherblog.blog.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

/**
 * 媒体版本实体
 *
 * @ref 媒体库深度优化方案 - Phase 5: 协作与权限
 * @author AI Assistant
 * @since 2.5.0
 */
@Data
@Entity
@Table(name = "media_versions", indexes = {
    @Index(name = "idx_media_versions_file", columnList = "media_file_id"),
    @Index(name = "idx_media_versions_created", columnList = "created_at")
}, uniqueConstraints = {
    @UniqueConstraint(name = "uq_media_version", columnNames = {"media_file_id", "version_number"})
})
public class MediaVersion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * 所属媒体文件
     */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "media_file_id", nullable = false)
    private MediaFile mediaFile;

    /**
     * 版本号
     */
    @Column(name = "version_number", nullable = false)
    private Integer versionNumber;

    /**
     * 文件路径
     */
    @Column(name = "file_path", nullable = false, length = 500)
    private String filePath;

    /**
     * 访问URL
     */
    @Column(name = "file_url", nullable = false, length = 500)
    private String fileUrl;

    /**
     * 文件大小 (bytes)
     */
    @Column(name = "file_size", nullable = false)
    private Long fileSize;

    /**
     * 变更描述
     */
    @Column(name = "change_description", columnDefinition = "TEXT")
    private String changeDescription;

    /**
     * 创建者
     */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;
}

File: apps/server-go/internal/model/media_version.go
package model

import "time"

// MediaVersion 对应数据库 `media_versions` 表，记录媒体文件的历史版本信息。
// 每次替换文件内容时会创建一条新的版本记录，以支持版本回溯。
type MediaVersion struct {
	ID                int64      `db:"id"`
	MediaFileID       int64      `db:"media_file_id"`       // 所属媒体文件 ID（外键，关联 media_files）
	VersionNumber     int        `db:"version_number"`      // 版本号，从 1 开始递增
	FilePath          string     `db:"file_path"`           // 该版本文件的存储路径
	FileURL           string     `db:"file_url"`            // 该版本文件的公开访问 URL
	FileSize          int64      `db:"file_size"`           // 该版本文件大小（字节）
	ChangeDescription *string    `db:"change_description"`  // 本次版本变更的说明，可为空
	CreatedBy         *int64     `db:"created_by"`          // 创建该版本的用户 ID，可为空
	CreatedAt         *time.Time `db:"created_at"`          // 版本创建时间
}
